functions/snippets/src/test/java/com/example/functions/OcrTests.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.functions;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.testing.TestLogHandler;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.List;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class OcrTests {
+  private static String FUNCTIONS_BUCKET = "nodejs-docs-samples-tests";
+  private static String RESULT_BUCKET = System.getenv("RESULT_BUCKET");
+
+  private static final Logger PROCESS_IMAGE_LOGGER = Logger.getLogger(
+      OcrProcessImage.class.getName());
+  private static final Logger SAVE_RESULT_LOGGER = Logger.getLogger(OcrSaveResult.class.getName());
+  private static final Logger TRANSLATE_TEXT_LOGGER = Logger.getLogger(
+      OcrTranslateText.class.getName());
+
+  private static final TestLogHandler LOG_HANDLER = new TestLogHandler();
+
+  private static final Gson gson = new Gson();
+
+  private static final Storage storage = StorageOptions.getDefaultInstance().getService();
+
+  @BeforeClass
+  public static void beforeClass() {
+    PROCESS_IMAGE_LOGGER.addHandler(LOG_HANDLER);
+    SAVE_RESULT_LOGGER.addHandler(LOG_HANDLER);
+    TRANSLATE_TEXT_LOGGER.addHandler(LOG_HANDLER);
+  }
+
+  @After
+  public void afterTest() {
+    LOG_HANDLER.clear();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void functionsOcrProcess_shouldValidateParams() throws IOException {
+    new OcrProcessImage().accept(new GcsEvent(), null);
+  }
+
+  @Test
+  public void functionsOcrProcess_shouldDetectText() throws IOException {
+    GcsEvent gcsEvent = new GcsEvent();
+    gcsEvent.bucket = FUNCTIONS_BUCKET;
```

Comment:
Should use either a setter (or constructor) to set these fields.

Reply:
Done.